gui/wxpython/datacatalog/catalog.py

```diff
@@ -110,33 +113,141 @@ def OnCreateMapset(self, event):
+
+    def OnImportGdalLayers(self, event):
+        """Convert multiple GDAL layers to GRASS raster map layers"""
+        from modules.import_export import GdalImportDialog
+        dlg = GdalImportDialog(parent=self, giface=self.giface)
+        dlg.CentreOnScreen()
+        dlg.Show()
+
+    def OnImportOgrLayers(self, event):
         """Convert multiple OGR layers to GRASS vector map layers"""
         from modules.import_export import OgrImportDialog
         dlg = OgrImportDialog(parent=self, giface=self.giface)
         dlg.CentreOnScreen()
         dlg.Show()
 
-    def OnImportGdalLayers(self, event, cmd=None):
-        """Convert multiple GDAL layers to GRASS raster map layers"""
+    def OnUnpackRaster(self, event):
+        """Unpack raster map handler"""
+        GUI(parent=self, giface=self.giface).ParseCommand(cmd=['r.unpack'])
+
+    def OnUnpackVector(self, event):
+        """Unpack vector map handler"""
+        GUI(parent=self, giface=self.giface).ParseCommand(cmd=['v.unpack'])
+
+    def OnLinkGdalLayers(self, event):
+        """Link multiple GDAL layers to GRASS raster map layers"""
         from modules.import_export import GdalImportDialog
-        dlg = GdalImportDialog(parent=self, giface=self.giface)
+        dlg = GdalImportDialog(parent=self, giface=self.giface, link=True)
         dlg.CentreOnScreen()
         dlg.Show()
 
-    def OnCreateLocation(self, event):
-        """Create new location"""
-        db_node, loc_node, mapset_node = self.tree.GetCurrentDbLocationMapsetNode()
-        self.tree.CreateLocation(db_node)
+    def OnLinkOgrLayers(self, event):
+        """Links multiple OGR layers to GRASS vector map layers"""
+        from modules.import_export import OgrImportDialog
+        dlg = OgrImportDialog(parent=self, giface=self.giface, link=True)
+        dlg.CentreOnScreen()
+        dlg.Show()
 
-    def OnDownloadLocation(self, event):
-        """Download location to current grass database"""
-        db_node, loc_node, mapset_node = self.tree.GetCurrentDbLocationMapsetNode()
-        self.tree.DownloadLocation(db_node)
+    def OnRasterOutputFormat(self, event):
+        """Set raster output format handler"""
+        from modules.import_export import GdalOutputDialog
+        dlg = GdalOutputDialog(parent=self, ogr=False)
+        dlg.CentreOnScreen()
+        dlg.Show()
+
+    def OnVectorOutputFormat(self, event):
+        """Set vector output format handler"""
+        from modules.import_export import GdalOutputDialog
+        dlg = GdalOutputDialog(parent=self, ogr=True)
+        dlg.CentreOnScreen()
+        dlg.Show()
+
+    def OnImportRasterAscii(self, event):
+        """Create raster map from x,y,z data handler"""
+        GUI(parent=self, giface=self.giface).ParseCommand(cmd=['r.in.xyz'])
+
+    def OnImportVectorAscii(self, event):
+        """Create vector map from x,y,z data handler"""
+        GUI(parent=self, giface=self.giface).ParseCommand(cmd=['v.in.ascii'])
```

Comment:
I would make a generic handler for these and the r/v.pack ones and use lambda function in Bind().

Reply:
Good point. I made the generic function GuiParseCommand which is then used in Bind() for all four menu items.